src/Fennel/CSharp.fs

```diff
@@ -33,7 +33,16 @@ type Comment(text : string) =
         member this.IsBlank = false
         
 type MetricTypeEnum = | Counter = 1 | Gauge = 2 | Histogram = 3 | Summary = 4 | Untyped = 0
-          
+type MetricTypeEnumHelpers() =
```

Comment:
Can I suggest the following change:

```suggestion
module MetricTypeEnum =
    let ToMetricType (metricType: MetricTypeEnum) =
        match metricType with
        | MetricTypeEnum.Counter -> Counter
        | MetricTypeEnum.Gauge -> Gauge
        | MetricTypeEnum.Histogram -> Histogram
        | MetricTypeEnum.Summary -> Summary
        | MetricTypeEnum.Untyped -> Untyped
        | _ -> failwith "Invalid MetricTypeEnum %A" metricType
        
    let FromMetricType (metricType: MetricType) =
        match metricType with
        | Counter -> MetricTypeEnum.Counter
        | Gauge -> MetricTypeEnum.Gauge
        | Histogram -> MetricTypeEnum.Histogram
        | Summary -> MetricTypeEnum.Summary
        | Untyped -> MetricTypeEnum.Untyped
```

Then line 48 `member this.MetricType = metricType |> function` etc.  can be replaced by `member this.MetricType = metricType |> MetricTypeEnum.FromMetricType`.

While you are improving this, I was thinking that maybe it is worth adding a constructor to `MetricType`, so you do not need to know about the FSharp type `Fennel.MetricType` at all if you wanted to new up a  `MetricType` line. 

So there would be an extra constructor like this? What do you think? This way you could use the enum to construct it if you needed an instance.
```fsharp
type MetricType(metricName : MetricName, metricType : Fennel.MetricType) =
    new (metricName : MetricName, metricTypeEnum : MetricTypeEnum) =   
        MetricType(metricName, MetricTypeEnum.ToMetricType(metricTypeEnum))
```